Compiler warnings: writing 1 byte into a region of size 0

After I upgraded my Ubuntu distro to Ubuntu 22, I got the following warnings when compiling:

```
In file included from /home/jeff/forks/monero/external/boost/archive/portable_binary_oarchive.hpp:32,
                 from /home/jeff/forks/monero/src/p2p/net_peerlist.cpp:37:
In function ‘void boost::archive::reverse_bytes(signed char, char*)’,
    inlined from ‘void boost::archive::portable_binary_oarchive::save_impl(intmax_t, char)’ at /home/jeff/forks/monero/external/boost/archive/portable_binary_oarchive.hpp:267:26:
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;

```


Here's my system info:

```
OS: Ubuntu 22
g++: (Ubuntu 11.2.0-19ubuntu1) 11.2.0
CPU: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
```

Here is another trace:

```
In function ‘void boost::archive::reverse_bytes(signed char, char*)’,
    inlined from ‘void boost::archive::portable_binary_iarchive::load_impl(intmax_t&, char)’ at /home/jeff/forks/monero/external/boost/archive/portable_binary_iarchive.hpp:265:18,
    inlined from ‘void boost::archive::portable_binary_iarchive::load(T&) [with T = boost::serialization::collection_size_type]’ at /home/jeff/forks/monero/external/boost/archive/portable_binary_iarchive.hpp:108:18,
    inlined from ‘static void boost::archive::load_access::load_primitive(Archive&, T&) [with Archive = boost::archive::portable_binary_iarchive; T = boost::serialization::collection_size_type]’ at /usr/include/boost/archive/detail/iserializer.hpp:108:16,
    inlined from ‘static void boost::archive::detail::load_non_pointer_type<Archive>::load_primitive::invoke(Archive&, T&) [with T = boost::serialization::collection_size_type; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/iserializer.hpp:384:40,
    inlined from ‘static void boost::archive::detail::load_non_pointer_type<Archive>::invoke(Archive&, T&) [with T = boost::serialization::collection_size_type; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/iserializer.hpp:461:22,
    inlined from ‘void boost::archive::load(Archive&, T&) [with Archive = boost::archive::portable_binary_iarchive; T = boost::serialization::collection_size_type]’ at /usr/include/boost/archive/detail/iserializer.hpp:624:18,
    inlined from ‘void boost::archive::detail::common_iarchive<Archive>::load_override(T&) [with T = boost::serialization::collection_size_type; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/common_iarchive.hpp:67:22,
    inlined from ‘void boost::archive::portable_binary_iarchive::load_override(T&) [with T = boost::serialization::collection_size_type]’ at /home/jeff/forks/monero/external/boost/archive/portable_binary_iarchive.hpp:160:52,
    inlined from ‘Archive& boost::archive::detail::interface_iarchive<Archive>::operator>>(T&) [with T = boost::serialization::collection_size_type; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/interface_iarchive.hpp:68:36,
    inlined from ‘void boost::serialization::load(Archive&, boost::serialization::nvp<T>&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = boost::serialization::collection_size_type]’ at /usr/include/boost/serialization/nvp.hpp:56:8,
    inlined from ‘static void boost::serialization::free_loader<Archive, T>::invoke(Archive&, T&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = boost::serialization::nvp<boost::serialization::collection_size_type>]’ at /usr/include/boost/serialization/split_free.hpp:58:13,
    inlined from ‘void boost::serialization::split_free(Archive&, T&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = boost::serialization::nvp<boost::serialization::collection_size_type>]’ at /usr/include/boost/serialization/split_free.hpp:74:18,
    inlined from ‘void boost::serialization::serialize(Archive&, boost::serialization::nvp<T>&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = boost::serialization::collection_size_type]’ at /usr/include/boost/serialization/nvp.hpp:65:15,
    inlined from ‘void boost::serialization::serialize_adl(Archive&, T&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = boost::serialization::nvp<boost::serialization::collection_size_type>]’ at /usr/include/boost/serialization/serialization.hpp:118:14,
    inlined from ‘static void boost::archive::detail::load_non_pointer_type<Archive>::load_only::invoke(Archive&, const T&) [with T = boost::serialization::nvp<boost::serialization::collection_size_type>; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/iserializer.hpp:395:48,
    inlined from ‘static void boost::archive::detail::load_non_pointer_type<Archive>::invoke(Archive&, T&) [with T = const boost::serialization::nvp<boost::serialization::collection_size_type>; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/iserializer.hpp:461:22,
    inlined from ‘void boost::archive::load(Archive&, T&) [with Archive = boost::archive::portable_binary_iarchive; T = const boost::serialization::nvp<boost::serialization::collection_size_type>]’ at /usr/include/boost/archive/detail/iserializer.hpp:624:18,
    inlined from ‘void boost::archive::detail::common_iarchive<Archive>::load_override(T&) [with T = const boost::serialization::nvp<boost::serialization::collection_size_type>; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/common_iarchive.hpp:67:22,
    inlined from ‘void boost::archive::portable_binary_iarchive::load_override(T&) [with T = const boost::serialization::nvp<boost::serialization::collection_size_type>]’ at /home/jeff/forks/monero/external/boost/archive/portable_binary_iarchive.hpp:160:52,
    inlined from ‘Archive& boost::archive::detail::interface_iarchive<Archive>::operator>>(T&) [with T = const boost::serialization::nvp<boost::serialization::collection_size_type>; Archive = boost::archive::portable_binary_iarchive]’ at /usr/include/boost/archive/detail/interface_iarchive.hpp:68:36,
    inlined from ‘void boost::serialization::load(Archive&, std::vector<U, Allocator>&, unsigned int, mpl_::false_) [with Archive = boost::archive::portable_binary_iarchive; U = crypto::public_key; Allocator = std::allocator<crypto::public_key>]’ at /usr/include/boost/serialization/vector.hpp:84:8,
    inlined from ‘void boost::serialization::load(Archive&, std::vector<U, Allocator>&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; U = crypto::public_key; Allocator = std::allocator<crypto::public_key>]’ at /usr/include/boost/serialization/vector.hpp:165:9,
    inlined from ‘static void boost::serialization::free_loader<Archive, T>::invoke(Archive&, T&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = std::vector<crypto::public_key>]’ at /usr/include/boost/serialization/split_free.hpp:58:13,
    inlined from ‘void boost::serialization::split_free(Archive&, T&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = std::vector<crypto::public_key>]’ at /usr/include/boost/serialization/split_free.hpp:74:18,
    inlined from ‘void boost::serialization::serialize(Archive&, std::vector<U, Allocator>&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; U = crypto::public_key; Allocator = std::allocator<crypto::public_key>]’ at /usr/include/boost/serialization/vector.hpp:176:37,
    inlined from ‘void boost::serialization::serialize_adl(Archive&, T&, unsigned int) [with Archive = boost::archive::portable_binary_iarchive; T = std::vector<crypto::public_key>]’ at /usr/include/boost/serialization/serialization.hpp:118:14,
    inlined from ‘void boost::archive::detail::iserializer<Archive, T>::load_object_data(boost::archive::detail::basic_iarchive&, void*, unsigned int) const [with Archive = boost::archive::portable_binary_iarchive; T = std::vector<crypto::public_key>]’ at /usr/include/boost/archive/detail/iserializer.hpp:187:40:
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:48:15: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   48 |         *last = *first;
      |         ~~~~~~^~~~~~~~
/home/jeff/forks/monero/external/boost/archive/portable_binary_archive.hpp:49:16: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
   49 |         *first = x;
      |         ~~~~~~~^~~

```

--- a/external/boost/archive/portable_binary_archive.hpp
+++ b/external/boost/archive/portable_binary_archive.hpp
@@ -44,9 +44,12 @@ reverse_bytes(signed char size, char *address){
     char * first = address;
     char * last = first + size - 1;
     for(;first < last;++first, --last){
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-overflow="
         char x = *last;
         *last = *first;
         *first = x;
+#pragma GCC diagnostic pop
     }
 }
 

